.github/workflows/static-binary.yaml

```diff
@@ -54,6 +68,10 @@ jobs:
         name: "${{ steps.create_paths.outputs.artifact_name }}"
         path: "${{ env.BUILD_DIR }}/${{ steps.create_paths.outputs.artifact_name }}"
 
+    - name: Upload Artifact to GCS
+      run: |
```

Comment:
Should this run on `master` pushes only?

Reply:
We upload the artifact to GitHub regardless of the branch. Keep in mind that this workflow only runs for pushes to master and changes to the workflow file. I'm in favor of keeping it as it is, but I also don't have a strong opinion on this. We should unify it, though, for GitHub and GCS uploads.

Reply:
I'm fine with this uploading for all triggers if we have a way to distinguish the uploaded files. As is this looks like it overrides files with the same name.

Reply:
yes it does. but file names have git SHAs in them, so they should not clash.